src/main/java/com/redhat/devtools/lsp4ij/features/diagnostics/LSPDiagnosticHandler.java

```diff
@@ -93,7 +94,10 @@ private void updateDiagnostics(@NotNull PublishDiagnosticsParams params, @NotNul
             // Trigger Intellij validation to execute
             // {@link LSPDiagnosticAnnotator}.
             // which translates LSP Diagnostics into Intellij Annotation
-            DaemonCodeAnalyzer.getInstance(project).restart(psiFile);
+            Editor editor = LSPIJUtils.editorForElement(psiFile);

```

Comment:
No need to restart the analyzer if the file isn't in an editor tab, and that will be the case for many files during a bulk inspection run if we stick with reacting to published diagnostic notifications.